### PR TITLE
Add glossary term refs for Thread, Matter, and Thread border router

### PR DESCRIPTION
**Pull request by c0ffeeca7**
## Proposed change

Add glossary term refs
- for the terms Thread, Matter, and Thread border router
- depends on #31054 

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/matter.markdown [hunk 1]
```diff
@@ -290,7 +290,7 @@ Also see this [extended troubleshooting guide](https://developers.home.google.co
 
 ### Unable to commission devices, it keeps giving errors or stops working randomly
 
-The Matter protocol relies on (local) IPv6 and mDNS (multicast traffic) which should be able to travel freely in your network. Matter devices that use Wi-Fi (including Thread Border routers) must be on the same LAN/VLAN as Home Assistant. Matter devices that only use {% term Thread %} must be joined to {% term Thread %} networks for which there is at least one border router connected to the Home Assistant LAN.
+The Matter protocol relies on (local) IPv6 and mDNS (multicast traffic) which should be able to travel freely in your network. Matter devices that use Wi-Fi (including {% term Thread border routers %}) must be on the same LAN/VLAN as Home Assistant. Matter devices that only use {% term Thread %} must be joined to {% term Thread %} networks for which there is at least one border router connected to the Home Assistant LAN.
```
Comment (MartinHjelmare) [inline source/_integrations/matter.markdown]: Does the term take plural (routers) too?

Comment (c0ffeeca7) [reply source/_integrations/matter.markdown]: right. that is the problem. Thank you!
#31050 needs to be merged before I can do this one

Comment (MartinHjelmare) [reply source/_integrations/matter.markdown]: Why is that PR needed for this PR?

Comment (c0ffeeca7) [reply source/_integrations/matter.markdown]: Haha, :see_no_evil: it doesn't. I thought I had mixed it in there. But I hadn't saved the file. So, the alias got its proper PR now. #31054. Thank you for bearing with me.

File: source/_integrations/matter.markdown [hunk 2]
```diff
@@ -283,7 +283,7 @@ Also see this [extended troubleshooting guide](https://developers.home.google.co
...
```
Comment (c0ffeeca7) [inline source/_integrations/matter.markdown]:
```suggestion
The Matter protocol relies on (local) IPv6 and mDNS (multicast traffic) which should be able to travel freely in your network. Matter devices that use Wi-Fi (including Thread border routers) must be on the same LAN/VLAN as Home Assistant. Matter devices that only use {% term Thread %} must be joined to {% term Thread %} networks for which there is at least one border router connected to the Home Assistant LAN.
```

File: source/_integrations/matter.markdown [hunk 3]
```diff
@@ -88,7 +88,7 @@ Running Matter on a Home Assistant Core installation is not supported.
...
 
-Note: The section below mentions third-party Thread border routers such as the Nest Hub v2 or the HomePod Mini. This doesn’t mean you have to add your devices into these ecosystems. Home Assistant only uses them to access the Thread radio network. The communication between the Home Assistant Matter controller and your Matter devices is encrypted. The Thread border router passes the data along, it cannot read its content.
+Note: The section below mentions third-party {% term Thread border routers %} such as the Nest Hub v2 or the HomePod Mini. This doesn’t mean you have to add your devices into these ecosystems. Home Assistant only uses them to access the Thread radio network. The communication between the Home Assistant Matter controller and your Matter devices is encrypted. The {% term Thread border router %} passes the data along, it cannot read its content.
```
Comment (c0ffeeca7) [inline source/_integrations/matter.markdown]:
```suggestion
Note: The section below mentions third-party Thread border routers such as the Nest Hub v2 or the HomePod Mini. This doesn’t mean you have to add your devices into these ecosystems. Home Assistant only uses them to access the Thread radio network. The communication between the Home Assistant Matter controller and your Matter devices is encrypted. The {% term Thread border router %} passes the data along, it cannot read its content.
```

**Review by c0ffeeca7 — COMMENTED** [review]
remove  plural term ref of Thread border routers

File: source/_integrations/homekit_controller.markdown [hunk 4]
```diff
@@ -116,28 +116,28 @@ You can add a HomeKit [compatible device](#supported-devices) to Home Assistant
 
 ## Adding a HomeKit device through Thread
 
-This section shows the the ways you can join a HomeKit device to a Thread network:
+This section shows the the ways you can join a HomeKit device to a {% term Thread %} network:
 
 1. via Home Assistant
-2. via Apple Thread border router
+2. via Apple {% term Thread border router %}
```
Comment (c0ffeeca7) [inline source/_integrations/homekit_controller.markdown]:
```suggestion
2. via Apple Thread border router
```

File: source/_integrations/homekit_controller.markdown [hunk 5]
```diff
@@ -146,34 +146,34 @@ This section describes how to add it via Home Assistant's preferred Thread netwo
...
+   - That's it. Your HomeKit device now communicates via {% term Thread %}.
 
 ### Adding a HomeKit device via Apple Thread border router
 
-There are two methods to add a HomeKit [compatible device](#supported-devices) to a Thread network:
+There are two methods to add a HomeKit [compatible device](#supported-devices) to a {% term Thread %} network:
 
 - via [Home Assistant's preferred Thread network](#adding-a-homekit-device-to-a-thread-network-via-home-assistant)
-- via Apple Thread border router
+- via Apple {% term Thread border router %}
```
Comment (c0ffeeca7) [inline source/_integrations/homekit_controller.markdown]:
```suggestion
- via Apple Thread border router
```

File: source/_integrations/homekit_controller.markdown [hunk 6]
```diff
@@ -163,17 +163,17 @@ This section describes how to add a HomeKit [compatible device](#supported-devic
 
 #### Prerequisites
 
-- An Apple device that can act as a Thread border router, such as a HomePod mini.
-- A HomeKit device which supports Thread. This is indicated by the Thread label on the packaging.
+- An Apple device that can act as a {% term Thread border router %}, such as a HomePod mini.
```
Comment (c0ffeeca7) [inline source/_integrations/homekit_controller.markdown]:
```suggestion
- An Apple device that can act as a Thread border router, such as a HomePod mini.
```

File: source/_integrations/matter.markdown [hunk 7]
```diff
@@ -88,7 +88,7 @@ Running Matter on a Home Assistant Core installation is not supported.
...
 
-Note: The section below mentions third-party Thread border routers such as the Nest Hub v2 or the HomePod Mini. This doesn’t mean you have to add your devices into these ecosystems. Home Assistant only uses them to access the Thread radio network. The communication between the Home Assistant Matter controller and your Matter devices is encrypted. The Thread border router passes the data along, it cannot read its content.
+Note: The section below mentions third-party Thread border routers such as the Nest Hub v2 or the HomePod Mini. This doesn’t mean you have to add your devices into these ecosystems. Home Assistant only uses them to access the Thread radio network. The communication between the Home Assistant Matter controller and your Matter devices is encrypted. The {% term Thread border router %} passes the data along, it cannot read its content.
```
Comment (c0ffeeca7) [inline source/_integrations/matter.markdown]:
```suggestion
Note: The section below mentions third-party Thread border routers such as the Nest Hub v2 or the HomePod Mini. This doesn’t mean you have to add your devices to these ecosystems. Home Assistant only uses them to access the Thread radio network. The communication between the Home Assistant Matter controller and your Matter devices is encrypted. The Thread border router passes the data along. It cannot read its content.
```

File: source/_integrations/matter.markdown [hunk 8]
```diff
@@ -113,11 +113,11 @@ Note: The section below mentions third-party Thread border routers such as the N
     - Have the latest version of the Home Assistant Companion app installed.
     - Have Google Home app installed on the Android.
     - We are not going to add the new device to Google Home. The app is needed because Google included the Matter SDK there.
-    - If you are using {% term Thread %}: Make sure there is a Thread border router device (Nest Hub v2 or Nest Wi-Fi Pro) present in your home network.
+    - If you are using {% term Thread %}: Make sure there is a {% term Thread border router %} device (Nest Hub v2 or Nest Wi-Fi Pro) present in your home network.
```
Comment (c0ffeeca7) [inline source/_integrations/matter.markdown]:
```suggestion
    - If you are using {% term Thread %}: Make sure there is a Thread border router device (Nest Hub v2 or Nest Wi-Fi Pro) present in your home network.
```

File: source/_integrations/matter.markdown [hunk 9]
```diff
@@ -113,11 +113,11 @@ Note: The section below mentions third-party Thread border routers such as the N
...
   - iPhone
     - Have the iOS version 16 or higher
     - Have the latest version of the Home Assistant Companion app installed.
-    - If you are using {% term Thread %}: Make sure there is a Thread border router device (HomePod Mini or V2, Apple TV 4K) present in your home network.
+    - If you are using {% term Thread %}: Make sure there is a {% term Thread border router %} device (HomePod Mini or V2, Apple TV 4K) present in your home network.
```
Comment (c0ffeeca7) [inline source/_integrations/matter.markdown]:
```suggestion
    - If you are using {% term Thread %}: Make sure there is a Thread border router device (HomePod Mini or V2, Apple TV 4K) present in your home network.
```

File: source/_integrations/matter.markdown [hunk 10]
```diff
@@ -257,7 +257,7 @@ NOTE for Android users: You need to follow the instructions at the bottom of the
 
 - Using Thread-based Matter devices in Home Assistant requires Home Assistant OS version 10 and above. Not using Home Assistant OS is at your own risk. We do provide some [documentation](https://github.com/home-assistant-libs/python-matter-server/blob/main/README.md) on how to run the Matter Server as a Docker container. The documentation includes a description of the host and networking requirements.
 
-- To use {% term Thread %} devices you will need a {% term Thread %} network with at least one Thread Border Router in your network nearby the {% term Thread %} device(s). Apple users need for example the Apple TV 4K or the HomePod Mini, while Google users need a Nest Hub V2. Use the Thread integration in Home Assistant to diagnose your {% term Thread %} network(s).
+- To use {% term Thread %} devices you will need a {% term Thread %} network with at least one {% term Thread border router %} in your network nearby the {% term Thread %} device(s). Apple users need for example the Apple TV 4K or the HomePod Mini, while Google users need a Nest Hub V2. Use the Thread integration in Home Assistant to diagnose your {% term Thread %} network(s).
```
Comment (c0ffeeca7) [inline source/_integrations/matter.markdown]:
```suggestion
- To use {% term Thread %} devices you will need a {% term Thread %} network with at least one Thread border router in your network nearby the {% term Thread %} device(s). Apple users, for example, need the Apple TV 4K or the HomePod Mini, while Google users need a Nest Hub V2. Use the Thread integration in Home Assistant to diagnose your {% term Thread %} network(s).
```

**Review by MartinHjelmare — APPROVED** [review]

**Comment by c0ffeeca7** [comment]
Thank you!

**Pull request merged**